### PR TITLE
feat(siren): Add StreetSiren Double Deck HTS temperature support

### PR DESCRIPTION
**Pull request by Taknok**
## Summary
This PR adds support for StreetSiren DoubleDeck devices to the existing HTS-based internal temperature handling path, by including street_siren_double_deck and street_siren_double_deck_fibra in the temperature device type sets used by the coordinator. It also refactors sensor creation so that temperature sensors are instantiated for these hub-device temperature families even when the initial gRPC snapshot does not yet contain device.statuses["temperature"].

The result is cleaner logic with a helper function for status-sensor creation, and it fixes the missing temperature sensor issue for DoubleDeck sirens by ensuring they are treated as HTS-sourced temperature devices rather than being ignored until a status value appears.

## Test plan

The modifications were successfully tested on my `street_siren_double_deck`.

- [x] `make check` passes locally on a freshly built Docker image
- [ ] New behaviour covered by unit tests in `tests/unit/`
- [x] Coverage stays at or above the 80% threshold
- [ ] User-facing strings updated in `strings.json` and the 14 translation files
- [ ] `README.md` / `CHANGELOG.md` updated when the change is user-visible

## Notes for the reviewer

- Added street_siren_double_deck and street_siren_double_deck_fibra to HUB_DEVICE_TEMPERATURE_DEVICE_TYPES and HTS_TEMPERATURE_DEVICE_TYPES.

- Ensures these siren variants are treated as HTS-sourced temperature devices

- Updated sensor setup so temperature entities are created for hub-device temperature types even before device.statuses["temperature"] is present

- Refactored sensor entity creation into _should_create_status_sensor() for cleaner readability

### CODE REVIEW

**Comment by bvis** [comment]
Thank you for this — and specifically for the line *"tested on my `street_siren_double_deck`"*, which is the part I can't produce myself. #354 shipped the volume and duration controls for this family but deliberately left temperature out, and the reason was exactly that nobody with the hardware could confirm which source carried it. You've closed that gap.

**The direction is right and matches the existing design.** You found both sets and understood why there are two: adding the type to `HTS_TEMPERATURE_DEVICE_TYPES` sources the value from HTS sub-key `0x02`, and adding it to `HUB_DEVICE_TEMPERATURE_DEVICE_TYPES` makes the merged value survive a stream snapshot that lacks it, while excluding the device from the slower gRPC fetch. That's the same treatment the other sirens get, for the same documented reason. No objection to any of that.

Four things before it can go in, one of which I'd like your opinion on rather than just a change.

### 1. The family has three variants and this covers two

`const.py` lists all three in `SIREN_DEVICE_TYPES`:

```
"street_siren_double_deck",
"street_siren_s_double_deck",      <- missing from both sets in this PR
"street_siren_double_deck_fibra",
```

You've already extrapolated from the model you own to `street_siren_double_deck_fibra`, which I think is correct — same family, same protocol. `street_siren_s_double_deck` is the same inference, so leaving it out means an owner of that exact SKU still gets nothing, for no reason we could defend. Please add it to both sets.

### 2. `_should_create_status_sensor` reaches further than this PR intends

This is the one I'd like to talk about rather than dictate. The helper creates a temperature sensor for **every** member of `HUB_DEVICE_TEMPERATURE_DEVICE_TYPES`, not only the sirens you're fixing. That set has one member that is *not* HTS-sourced:

```
motion_protect_curtain_outdoor_mini
```

The Mini is the single family fetched over gRPC instead of HTS, and we have never confirmed that path actually yields a temperature for it — #269 is open precisely because no Mini sample has ever been captured. So on current evidence this change would create a Temperature entity for Mini owners that sits at `unknown` indefinitely. Home Assistant doesn't remove entities that stop being produced, so that's not self-correcting, and #354's changelog took the opposite position deliberately: *"creating entities it could never fill in would have been worse than leaving them out."*

I don't think you need to abandon the refactor — the underlying observation is correct and worth having, because a value that arrives later via HTS shouldn't require the entity to already exist. It just wants a narrower gate. Something like keying on the HTS set, which is the one that guarantees a source:

```python
if key == "temperature" and device.device_type in HTS_TEMPERATURE_DEVICE_TYPES:
    return True
return key in device.statuses
```

That covers all your sirens and leaves the Mini exactly as it is today. If you think the Mini genuinely does report over gRPC and I'm being over-cautious, say so — you've been closer to this hardware than I have.

### 3. CI is red

Both failures are in `sensor.py` and are mechanical:

- **lint** — the new `from custom_components.aegis_ajax.const import …` is out of order; it belongs before the `coordinator` import.
- **format-check** — the nested `def _should_create_status_sensor` needs a blank line before it.

`ruff check --fix . && ruff format .` inside the dev image clears both.

### 4. A unit test

Your own checklist has this unticked, so you're already aware. One test asserting a temperature sensor is created for a DoubleDeck whose `statuses` has no `temperature` yet would pin the behaviour — and a second asserting nothing is created for the type your gate excludes would document the decision from point 2. There are existing patterns in `tests/unit/test_sensor.py`.

### On timing

To set expectations honestly: I'm mid-release-train on `1.15.1` and holding it for field confirmation of an unrelated push fix, so I don't want to add a new device-behaviour change to that train. **Consider this accepted in principle** — once the points above are addressed I'll merge it into the first beta after `1.15.1` goes stable, rather than leaving it sitting.

If you'd rather I made the four changes myself on top of your branch, say so and I will; the authorship stays yours either way. And if you have a `street_siren_s_double_deck` or know someone who does, a confirmation on that SKU would turn point 1 from inference into an observation.

**Comment by bvis** [comment]
**`1.15.1` shipped as stable today, so the hold described above is lifted** — this PR is next in line and goes out in the first beta after it, as promised.

The four review points above are all that stands between here and that merge. The offer also stands: if you'd rather I made those changes on top of your branch, say so and I will — authorship stays yours either way.

**Comment by Taknok** [comment]
Hi,

Sorry for the late reply - it was a busy day.

First of all, thank you for your amazing integration and for taking the time to review my PR in such detail.

1. OK, I thought I was already being a bit overconfident by including the `_fibra` device, but I noticed that throughout the code their behavior is always grouped together. I'm not sure about the `_s` devices, but you're in a much better position to know whether they behave the same as the regular models. In any case, adding it is a reasonable inference, so I'll update the code.

2. Good catch regarding the Mini. I hadn't considered that case. I'll fix it.

3. OK, that's a quick fix.

4. That one may take me a little longer (probably a few days). I'll read through the existing tests and follow the same structure to keep everything consistent.

Regarding the timing, that's perfectly fine. There's no hurry on my side. Thanks again for your time, the detailed feedback, and the quick review!

**Comment by bvis** [comment]
Thanks for turning these around — all three Double Deck variants are in, and the gate now keys on the HTS set exactly as discussed. I took you up on the offer to finish it myself rather than send you another round, since what was left was mechanical. **Your authorship on the original commits is untouched**; I pushed one commit on top.

**The one real bug:** the helper imported `HTS_TEMPERATURE_DEVICE_TYPES` from `const`, but that set lives in `api/hts/hub_state` — `const` only holds `HUB_DEVICE_TEMPERATURE_DEVICE_TYPES`. Easy to miss given the names, and it's what broke `type-check` and both `test` jobs (every test importing the sensor platform died on it). Fixed by importing from the right module, alongside the two sets `sensor.py` already pulls from there.

**Tests for the checklist item you left open**, six in total:

- One per Double Deck variant, asserting the temperature entity is created when `statuses` is still empty — which is the actual behaviour change. I checked these fail against the old `key in device.statuses` gate, so they pin the fix rather than just passing.
- One asserting the **Curtain Outdoor Mini stays excluded**. It is the only member of `HUB_DEVICE_TEMPERATURE_DEVICE_TYPES` with no HTS source, so a wider gate would create a sensor that never gets a value and that Home Assistant never evicts. This test is really there to document the decision so nobody widens the gate later without noticing.
- One asserting the Mini *does* get its sensor once a real value arrives, so it's clear the exclusion is about timing rather than dropping the family.
- One asserting the gate doesn't leak to `humidity` / `co2` / `signal_strength`.

**Also folded in:** current `main` merged (`1.15.1` shipped yesterday), a CHANGELOG entry under a new `1.16.0` section crediting you, and a README correction — the Supported Devices table said the Double Decks report no internal temperature, which your work makes false.

Full suite green locally: 2108 tests, 90.11% coverage. Once CI confirms, this goes into the first beta after `1.15.1`, as promised.

One last thing, purely optional: if you ever get a reading you can sanity-check against a thermometer, it'd be good to know the value looks plausible and not, say, offset or scaled. Nothing blocks on it.

**Comment by bvis** [comment]
Merged — CI green across the board. Thanks @Taknok.

To be clear about what your contribution actually settled: `1.15.1` shipped the Double Deck volume and duration controls but left temperature out **specifically because nobody with the hardware could confirm which channel carried it**, and guessing would have meant creating a sensor that might never fill in. You owned the hardware, found both device-type lists, worked out why there are two, and tested it on the real siren. That is the part I could not do from here.

The second-order fix matters as much as the sirens, and it came out of your observation rather than mine: a value that arrives later over the status stream shouldn't require the entity to already exist. That reasoning now applies to the whole family, not just your model.

This goes out in the first beta after `1.15.1` and I'll ping you here when it's published so you can confirm it on your siren from a real install rather than a patched checkout.

**Comment by Taknok** [comment]
Thank you for the fixes, the detailed test coverage, and for rebasing onto 1.15.1.

Regarding the temperature readings, I've been monitoring them for the past three days, and they match the values reported by the official Ajax app.

As with any thermometer exposed to direct sunlight, the reported temperature rises well above the ambient air temperature. When the siren is in the shade, however, the readings seem realistic to other thermometers and within the expected accuracy for a non-calibrated sensor.

**Comment by Taknok** [comment]
Ok, just saw that we cross-post comments. I will implement the tests in another PR (to keep code clean) if you want.

Edit: Ok I just saw that you implemented the tests. Thank you !

**Comment by bvis** [comment]
@Taknok — **`1.16.0-beta.1` is published** and carries your change: https://github.com/bvis/aegis-hass/releases/tag/v1.16.0-beta.1

HACS → the integration → three-dot menu → Redownload → tick "Show beta versions" → pick `1.16.0-beta.1`, then restart Home Assistant.

Your Double Deck should gain an **internal temperature** sensor. It appears at startup and fills in on the first status update rather than showing a value immediately, which is the timing fix described above working as intended.

Whenever you get a chance, the one thing worth knowing is whether the number is plausible for where the siren is mounted. No rush, and no need for anything precise — "it says 24 °C and that's about right for an outdoor unit in the shade" is enough. If it looks off by a constant or a factor, that's exactly the kind of thing only you can spot.

**Comment by Taknok** [comment]
Hi,
Here is the live test result:
PXL_20260805_084757823 MP
Screenshot_20260805-104818
Screenshot_20260805-105101

Thank you !

PS: I will PR for motion cam phod later.

**Comment by Taknok** [comment]
Currently temperature is 31°C.
image

Ajax app:
Screenshot_20260806-001521


Here are the logs:
```
2026-08-06 00:11:27.058 DEBUG (MainThread) [custom_components.aegis_ajax.api.hts.client] Hub 0029B936: STATUS_BODY non-hub devices (#123 probe): 9DD86D26=[0x01=01,0x02=21,0x03=03,0x04=00,0x05=0064,0x06=0000,0x07=80000000,0x0b=01,0x0c=01,0x0e=80,0x0f=00,0x10=00,0x17=00000000,0x2c=01,0x31=00,0x32=01,0x39=01,0x3a=00,0x46=00,0x47=00,0x99=1f,0x9f=00,0xb7=00,0xc2=80,0xc3=00,0xc6=00,0xf5=00,0xf7=00,0xf9=80], 4814CF17=[], 00B0BF9A=[0x02=1e,0x03=03,0x04=00,0x05=0064,0x06=0000,0x07=80000000,0x0b=01,0x0c=01,0x0e=80,0x0f=00,0x10=00,0x17=00000000,0x2c=01,0x31=01,0x32=00,0x99=1e,0x9f=00,0xb7=00,0xc2=80,0xc3=00,0xc6=00,0xf5=00,0xf7=00,0xf9=80], 3213EE90=[], 305918C1=[0x02=1e,0x03=03,0x04=00,0x05=0064,0x06=0000,0x07=80000000,0x0b=01,0x0c=01,0x0e=80,0x0f=00,0x10=00,0x17=00000000,0x2c=01,0x37=00,0x99=1c,0x9f=00,0xb7=00,0xc2=80,0xc3=00,0xc6=00,0xf5=00,0xf7=00,0xf9=80], 308CE402=[0x02=1c,0x03=03,0x04=00,0x05=0064,0x06=0000,0x07=80000000,0x0b=01,0x0c=01,0x0e=80,0x0f=00,0x10=00,0x17=00000000,0x2c=01,0x34=00,0x42=01,0x4b=03,0x52=00,0x5a=00,0x5b=01,0x5f=00,0x87=00,0x8e=80,0x99=3b,0x9f=00,0xb7=00,0xc2=00,0xc3=01,0xc6=00,0xf5=00,0xf7=00,0xf9=80], 002639CE=[0x02=80,0x03=03,0x04=00,0x05=0064,0x06=0000,0x07=80000000,0x0b=01,0x0c=01,0x0e=80,0x0f=00,0x10=00,0x17=00000000,0x2c=01,0x31=01,0x33=00,0x34=00,0x35=00,0x37=00,0x99=8a,0x9f=00,0xb7=00,0xc2=80,0xc3=00,0xc6=00,0xf5=00,0xf7=00,0xf9=80], 00000001=[], 3050A2A7=[0x02=1f,0x03=03,0x04=00,0x05=0064,0x06=0000,0x07=80000000,0x0b=01,0x0c=01,0x0e=80,0x0f=00,0x10=00,0x17=00000000,0x2c=01,0x36=00,0x37=00,0x43=80,0x79=80,0x7d=01,0x99=37,0x9f=00,0xb7=00,0xc2=80,0xc3=00,0xc6=00,0xf5=00,0xf7=00,0xf9=80], 00000002=[], 305654EA=[0x02=1e,0x03=03,0x04=00,0x05=0064,0x06=0000,0x07=80000000,0x0b=01,0x0c=01,0x0e=80,0x0f=00,0x10=00,0x17=00000000,0x2c=01,0x31=00,0x32=00,0x39=01,0x3a=00,0x46=00,0x47=00,0x99=1e,0x9f=00,0xb7=00,0xc2=80,0xc3=00,0xc6=00,0xf5=00,0xf7=00,0xf9=80], 00000003=[], 00000001=[0x02=01,0x18=80], 00000002=[0x02=02], 00ABE560=[0x02=1f,0x03=03,0x04=00,0x05=0064,0x06=0000,0x07=80000000,0x0b=01,0x0c=01,0x0e=80,0x0f=00,0x10=00,0x17=00000000,0x2c=01,0x31=01,0x32=00,0x99=1e,0x9f=00,0xb7=00,0xc2=80,0xc3=00,0xc6=00,0xf5=00,0xf7=00,0xf9=80], 308CB897=[0x02=1e,0x03=03,0x04=00,0x05=0064,0x06=0000,0x07=80000000,0x0b=01,0x0c=01,0x0e=80,0x0f=00,0x10=00,0x17=00000000,0x2c=01,0x34=00,0x42=01,0x4b=03,0x52=00,0x5a=00,0x5b=01,0x5f=00,0x87=00,0x8e=80,0x99=3a,0x9f=00,0xb7=00,0xc2=00,0xc3=01,0xc6=00,0xf5=00,0xf7=00,0xf9=80], 307B3179=[0x02=22,0x03=03,0x04=00,0x05=0064,0x06=0000,0x07=80000000,0x0b=01,0x0c=01,0x0e=80,0x0f=00,0x10=00,0x17=00000000,0x2c=01,0x31=00,0x32=00,0x39=01,0x3a=00,0x46=00,0x47=00,0x99=1e,0x9f=00,0xb7=00,0xc2=80,0xc3=00,0xc6=00,0xf5=00,0xf7=00,0xf9=80], 305AB471=[0x02=80,0x03=80,0x04=80,0x05=0064,0x06=8000,0x07=80000000,0x0b=01,0x0c=01,0x0e=80,0x0f=00,0x10=00,0x17=00000000,0x2c=80,0x99=1b,0x9f=00,0xb7=00,0xc2=80,0xc3=00,0xc6=00,0xf5=00,0xf7=00,0xf9=80], 304FB6A3=[0x02=20,0x03=03,0x04=00,0x05=0064,0x06=0000,0x07=80000000,0x0b=01,0x0c=01,0x0e=80,0x0f=00,0x10=00,0x17=00000000,0x2c=01,0x31=00,0x32=00,0x39=01,0x3a=00,0x46=00,0x47=00,0x99=1e,0x9f=00,0xb7=00,0xc2=80,0xc3=00,0xc6=00,0xf5=00,0xf7=00,0xf9=80], 00000001=[], 5B4CCD2C=[], 432B35C8=[], 730454E3=[], 
```

in png:
image

**Comment by bvis** [comment]
With `1.16.0` stable out (https://github.com/bvis/aegis-hass/releases/tag/v1.16.0), closing the loop here properly: your 31 °C reading with the HA and Ajax app screenshots side by side was exactly the confirmation this feature was gated on — thank you, and apologies it went unacknowledged for a few days. Your Double Deck is why the whole siren family's temperature handling now creates entities from family membership instead of from a value happening to be present, which fixed a latent gap well beyond your own hardware.
